webpack/scenes/ContentViews/Details/Filters/CVErrataDateFilterContent.js

```diff
@@ -83,13 +84,24 @@ const CVErrataDateFilterContent = ({
     setEndDate(convertAPIDateToUIFormat(ruleEndDate));
     setSelectedTypes(types);
     setDateType(ruleDateType);
+    setAllowOtherTypes(ruleAllowOtherTypes);
   };
 
   const onTypeSelect = (selection) => {
+    let updatedSelectedTypes;
     if (selectedTypes.includes(selection)) {
+      // If the selection is the only selection remaining, do not allow it to be removed
       if (selectedTypes.length === 1) return;
-      setSelectedTypes(selectedTypes.filter(e => e !== selection));
-    } else setSelectedTypes([...selectedTypes, selection]);
+
+      // Filter out the current selection from the selected types to update
+      updatedSelectedTypes = selectedTypes.filter(e => e !== selection);
+    } else {
+      updatedSelectedTypes = [...selectedTypes, selection];
+    }
+
+    // If 'other' is selected, update the allowOtherTypes field
+    setSelectedTypes(updatedSelectedTypes.filter(e => e !== 'other'));
+    setAllowOtherTypes(updatedSelectedTypes.includes('other'));
```

Comment:
The filter in line 103 is what's causing the weird behavior of the checkboxes. But the good news is, there's no need for that filter. Let me explain..

Remember that in React, state stays inside the component. Just because you have a state called `allowOtherTypes` doesn't mean it will magically be bound to some data field and updated when it changes. (That's closer to how AngularJS behaves.) All state does is cause its component to rerender when its value is updated.

I see these two states here:
```
const [selectedTypes, setSelectedTypes] = useState(types);
const [allowOtherTypes, setAllowOtherTypes] = useState(ruleAllowOtherTypes);
```

State describes what's happening currently within the component. I don't think we need the `allowOtherTypes` state at all. "Other" can just be a selected type just like any other. 

Where we need to handle it is above, in the `onSave` method. In there, we can normalize the data, e.g. see if `selectedTypes` include "other" and then transform that into the `allow_other_types` param -- but _only_ when we actually send the data to the backend. Before then, it doesn't matter.

So my suggestion is
1. Revert all the refactoring from the `onTypeSelect` method - it can go back to what it was doing before.
2. In the `onSave` method, see if `selectedTypes` includes `other`. If it does, remove it from `selectedTypes` and instead send the `allow_other_types` param. You'll want to add this to the object right after `types: selectedTypes,`.